### PR TITLE
Fix issue 1360

### LINKED ISSUE #1360: Greetings is not render(update) when authState is changed
## Do you want to request a feature or report a bug?
Bug

## What is the current behavior?
1. I wrote below codes.
```
import React from "react";
import { ConfirmSignUp, ForgotPassword, SignIn, SignUp, VerifyContact, Authenticator, Greetings } from 'aws-amplify-react';

// signed in view at my apps
class MyView extends React.Component {
  constructor (props) {
    super(props);
    this.state = {
      myState: null,
    };
  }
    
  render () {
    if (this.props.authState === 'signedIn') {
      return (
        <div>
          <h2>State is ::</h2>
          <p>{this.props.authState}</p>
        </div>
      );
    } else {
      return(<p></p>);
    }
  }
}

export class MyComp extends React.Component {
  handleAuthStateChange(state) {
      if (state === 'signedIn') {
          // do something
      }
  }
  render () {
    return (
      <Authenticator hideDefault={true} onStateChange={this.handleAuthStateChange}>
        <SignIn />
        <VerifyContact />
        <SignUp />
        <ConfirmSignUp />
        <ForgotPassword />
        <Greetings
          inGreeting={(username) => 'Hello ' + username}
          outGreeting="Please sign in..."
        />
        <MyView />
      </Authenticator>
    );
  }
}
```
2. When I sign-in at first, Greetings shows `Please sign in...`. I expected it. no problem.

3. However when authState is `signedIn`, `Please sign in...` is still displayed and not shows sign out button.

4. This problem is cleared by browser reload, and turn to `Hello username` and shows signout button.

5. When I signout, <Authenticator/> shows signIn componnent, but Greetings shows `Hello username`.

6. The problem of 5, I reload browser, turn to `Please sign in...`.

## What is the expected behavior?

In, `what is current behavior` section,

3. When I sign-in, turn to `Hello username` without browser reload.

5. When I sign-out, turn to `Please sign-in` without browser reload.

### OS
- win10, win7, mac
- I think OS is not affected

### Browser
- Chrome 68.0.3440.84（Official Build）

### Did this work in previous versions?
- No

### My package.json
```
  "dependencies": {
    "@material-ui/core": "^1.3.0",
    "ajv": "^6.5.0",
    "ajv-keywords": "^3.2.0",
    "aws-amplify": "^1.0.2",
    "aws-amplify-react": "^1.0.2",
    "babel-polyfill": "^6.26.0",
    "prop-types": "^15.6.2",
    "react": "^16.4.0",
    "react-dom": "^16.4.0",
    "react-router-dom": "^4.3.1",
    "react-scripts": "1.1.4"
  },

```

### PR DESCRIPTION
**Pull request by powerful23**
*Issue #, if available:*
fixes #1360 
*Description of changes:*

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by jordanranz** [comment]
Does auth state need to be changed in the AuthStateWrapper too?

File: packages/aws-amplify-react/src/Auth/Authenticator.jsx [hunk 1]
```diff
@@ -44,7 +44,7 @@ export default class Authenticator extends Component {
         this.onHubCapsule = this.onHubCapsule.bind(this);
 
         this._initialAuthState = this.props.authState || 'signIn';
-        this.state = { auth: 'loading' };
```
Comment (jordanranz) [inline packages/aws-amplify-react/src/Auth/Authenticator.jsx]: Is authData initially undefined?

Comment (powerful23) [reply packages/aws-amplify-react/src/Auth/Authenticator.jsx]: Yes

**Comment by powerful23** [comment]
> Does auth state need to be changed in the AuthStateWrapper too?

Done.

**Pull request merged**